app/models/manageiq/providers/azure/cloud_manager/provision/cloning.rb

```diff
@@ -82,13 +80,22 @@ def prepare_for_clone_task
             :caching      => 'ReadWrite',
             :osType       => os
           }
-        },
-        :networkProfile  => {
-          :networkInterfaces => [{:id => nic_id}],
         }
       }
     }
 
+    # The -1 value is set in ProvisionWorkflow to distinguish between the
+    # desire for a new Public IP address vs a private IP.
+    #
+    if floating_ip
+      nic_id = associated_nic || create_nic(true)
```

Comment:
@djberg96 - In what scenario will `create_nic` on this line be called? Looks like this line will only be executed if floating_ip is not nil, which means a valid floating IP was selected and there is no need to create a new one.

Reply:
This was something the specs caught. It's possible that a public IP exists, but isn't actually associated with a NIC yet.

Reply:
Ah I see. So this means when the user selects a floating ip from the drop down that is not associated with a NIC  we create a new IP called "<dest_name>-PublicIp" which isn't what the user intended. 
This is what we have always done, I wonder should we at least add some info logging since we are doing something the user didn't intend?

Reply:
The other option is to just create the NIC dynamically if it's not associated with a NIC, and then associate it with the newly created NIC. Excluding them would be easier, though.

In any case, I think that's something that should be in a separate PR. I don't get the impression that people create public IP's independently of a NIC much in practice.

Reply:
yeah it has always been there so i think its fine if it is addressed in a separate PR.